esp8266/tcp.go

```diff
@@ -0,0 +1,97 @@
+package esp8266
+
+import (
+	"strconv"
+	"time"
+)
+
+const (
+	TCPMuxSingle   = 0
+	TCPMuxMultiple = 1
+
+	TCPTransferModeNormal      = 0
+	TCPTransferModeUnvarnished = 1
+)
+
+// ConnectTCPSocket creates a new TCP socket connection for the ESP8266.
+// Currently only supports single connection mode.
+func (d *Device) ConnectTCPSocket(addr, port string) error {
+	protocol := "TCP"
+	val := "\"" + protocol + "\",\"" + addr + "\"," + port
+	d.Set(TCPConnect, val)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Is it walkaround to make sure there is a data waiting in buffer?

Reply:
There is a short delay from when the AT command is sent to the ESP8266 and when it responds/is ready for the next command or data.

I picked this value just at random based on experimenting at 115200 baud serial rate. Probably somewhere in the datasheet is a more specific way to calculate this.

Reply:
That’s why asked, because didn’t found it at all in specs. Will dig into it